src/geo/leaflet/leaflet-layer-view-factory.js

```diff
@@ -1,24 +1,60 @@
+var _ = require('underscore');
 var log = require('cdb.log');
 var LeafletTiledLayerView = require('./leaflet-tiled-layer-view');
 var LeafletWMSLayerView = require('./leaflet-wms-layer-view');
 var LeafletPlainLayerView = require('./leaflet-plain-layer-view');
 var LeafletCartoDBLayerGroupView = require('./leaflet-cartodb-layer-group-view');
 var LeafletTorqueLayerView = require('./leaflet-torque-layer-view');
 var LeafletCartoDBWebglLayerGroupView = require('./leaflet-cartodb-webgl-layer-group-view');
+var TangramCartoCSS = require('tangram-cartocss');
+var RenderModes = require('../../vis/render-modes');
 
-var LayerGroupViewConstructor = function (layerGroupModel, mapModel, options) {
-  if (options.vector) {
-    return new LeafletCartoDBWebglLayerGroupView(layerGroupModel, mapModel);
+var MAX_NUMBER_OF_FEATURES_FOR_WEBGL = 10e4;
+
+var LayerGroupViewConstructor = function (layerGroupModel, nativeMap, mapModel, settingsModel) {
+  if (canMapBeRenderedClientSide(mapModel, settingsModel)) {
+    return new LeafletCartoDBWebglLayerGroupView(layerGroupModel, nativeMap);
+  }
+
+  return new LeafletCartoDBLayerGroupView(layerGroupModel, nativeMap);
+};
+
+var canMapBeRenderedClientSide = function (mapModel, settingsModel) {
+  var mapRenderMode = settingsModel.get('renderMode');
+
+  if (mapRenderMode === RenderModes.VECTOR && isWebGLSupported()) {
+    return true;
+  }
+
+  if (mapRenderMode === RenderModes.RASTER) {
+    return false;
   }
-  return new LeafletCartoDBLayerGroupView(layerGroupModel, mapModel);
+
+  // RenderModes.AUTO
+  var featureCount = mapModel.getFeatureCount();
+  return isWebGLSupported() &&
```

Comment:
When deciding wether to use Vector or Raster the following conditions must be satisfied in order for CartoDB.js to go for Vector:

- Browser must support WebGL. 
- Map must have stats about the estimated number of features.
- The estimated number of features must be smaller than 10e4 (100000).
- CartoCSS of all CartoDB layers should be supported by tangram-cartocss library.

Sounds good @donflopez, @javisantana?